src/QAT/qat/purr/integrations/tket.py

```diff
@@ -346,6 +347,28 @@ def optimize_circuit(circ, architecture, opts):
         return False
 
 
+def get_coupling_subgraphs(couplings):
```

Comment:
this needs its own set of tests

Reply:
Added

Reply:
@hamidelmaazouz A quick search says there isn't one, at least not one that wouldn't require adding additional dependencies for just this feature. If we find one it can be replaced in a future PR. :slightly_smiling_face: